src/sampleutil.cpp

```diff
@@ -17,14 +17,62 @@ typedef qint32 int32_t;
 // When changing, be carefull to not prevent the vectorizing
 // https://gcc.gnu.org/projects/tree-ssa/vectorization.html
 
+
+static inline bool useAlignedAlloc() {
+    // Will return true on MSVC 32 bit builds and falls for
+    // Linux 32 and 64 bit builds
+    return (sizeof(long double) == 8 && sizeof(CSAMPLE*) <= 8 &&
+            sizeof(CSAMPLE*) == sizeof(size_t));
+}
+
 // static
 CSAMPLE* SampleUtil::alloc(unsigned int size) {
-    // TODO(XXX) align the array
-    return new CSAMPLE[size];
+    // For optimal use of SSE registers, it is required to align
+    // the sample buffers to 16 Byte (128 bit) boundary
+
+    // Pointers returned by malloc are aligned for the largest scalar type,
+    // which is long double with usually 16 byte.
+    // An exception is MSVC X86 where long double is mapped to double.
+
+    // In case of sizeof(long double) = 8
+    // this code over allocates the requested buffer to be able to
+    // shift the returned pointer to a 16 byte alignment
+    // In the memory before, a pointer to the original
+    // malloced area is stored used to free the memory
+    // This code can be replaced by C11 <stdlib.h> aligned_alloc()
+    // or MSVC ::_aligned_malloc(size, alignment) and  ::_aligned_free(ptr);
+
+    if (useAlignedAlloc()) {
+        // We need to shift the alignment to 16
```

Comment:
Unfortunately it is not only one boost header. Copying a noter library is also not the best idea.
Adding boost dependency for this is also overdone.

If this builds on windows. Lets stick with the current version. Non of our current builds will use internal aligned allocator, which actually works by the way. The only build which uses _aligned_malloc is win 32.

Once we have c++11 there is a standard function for this. So there is no need to spend more time on this topic.